### ISSUE TITLE
Permissions / Multiple users

### ISSUE THREAD

**Issue by TheFlyingCorpse** [home-assistant/ui-schema#144]
Awesome work on the new UI, it looks fresh 👍 

Any thoughts on implementing permissions in the new UI? It would be really awesome if it became possible to grant different users different access in this.

**Comment by iantrich** [comment]
With the person component and user-roles finally coming into place, we should probably start the discussion of how this will look in Lovelace.

I think each user should have their own configuration, but we need to consider how they can be easily maintained as well, especially if read-only accounts can't maintain their own;
i.e. Should there be a mechanism to share views and/or cards between users.
e.g. If I'm the maintainer and am creating views for my kids, I don't want to have to edit 4 configurations when I want to add a single device to a common view/device.

There's a lot to discuss here, but now seems like the right time to start exploring these things.

@balloob @bramkragten @thomasloven @robbiet480 @zsarnett @maykar

Any input is welcome 😄

**Comment by maykar** [comment]
For sure, common views would be very helpful. This could just be a simple as adding a shared view option in view configuration.

![image](https://user-images.githubusercontent.com/25127328/54689131-51b88180-4af5-11e9-8f6d-935dec9cda69.png)

**Comment by bramkragten** [comment]
As an admin you see everything, and you can set a group/user per view/card a non-admin can only control the elements that are only for that user? If he adds something it is only for his user?
Maybe even allow/deny lists per element?

Or only allow an admin to control the UI?

This gets messy real quick...

**Comment by maykar** [comment]
I'm thinking of it more as having a common/shared view that only admin can edit, but would be visible to all.

**Comment by iantrich** [comment]
@maykar I 100% want unique views per user. E.g. wife wants to see her Spotify integration, not mine.
@bramkragten yeah, and is why I think this is not going to be an easy discussion or change 😄 You also realize that more likely than not this will require us to bring `id` back 🙈

**Comment by maykar** [comment]
YAML anchors in the add card list seems like it would work nicely. I would still love to see some sort of common view. This would keep me from having to manually edit each users view to include important additions.

**Comment by bramkragten** [comment]
Entities are covered by the backend right? We will be able to set per entity if a group can see or edit an entity in the backend? So we will not show it if a user doesn't have access.

**Comment by bramkragten** [comment]
YAML anchors won't survive conversion to JSON, so ones saved we will lose it.

**Comment by iantrich** [comment]
I was never clear why we saved the configuration as JSON 🤷‍♂️

**Comment by bramkragten** [comment]
Well, everything is in JSON anyway, so why convert it to YAML before/after saving?

**Comment by balloob** [comment]
YAML anchors are not user friendly. The UI is for the normal folks. 

There are two problems to solve, and we should not conflate them, instead focus at one problem at a time:
 - Allow having different Lovelace configs per user/device etc
 - Allow card reuse by having some general cards and allow "including" or "linking" them into a Lovelace config.

**Comment by balloob** [comment]
It's totally fine to implement the first without the second.

**Comment by iantrich** [comment]
lmao, listening to the podcast now discussing this exact same thing right now :smile:

**Comment by jnpetersen** [comment]
I agree with @iantrich that each person should have their own files, that way it helps keep things clean per user.

I'd think that for each user, if it needs to be updated, you'd select Configure UI like you do now, but it will prompt for admin user account to make changes.

The other idea would be that under the admin account (all administrator users if you have multiple would share the same views) you would create all views, then you can assign a view to individual or multiple users.
